modules/lti-service-impl/src/main/java/org/opencastproject/lti/service/impl/LtiServiceImpl.java

```diff
@@ -0,0 +1,497 @@
+            final List<WorkflowInstance> workflowInstances = workflows
+                    .applyWorkflowToLatestVersion(Collections.singleton(targetMpId), newWorkflow).toList();
+            if (workflowInstances.isEmpty()) {
+              throw new RuntimeException(
+                      String.format("couldn't start workflow '%s' for event %s", publishWorkflowName, targetMpId));
+            }
+          } catch (WorkflowDatabaseException e) {
+            logger.error(String.format("couldn't instantiate workflow '%s'", publishWorkflowName), e);
+          } catch (NotFoundException e) {
+            logger.error(String.format("couldn't find media package while starting workflow workflow '%s'",
+                    publishWorkflowName), e);
+          }
+        }
+      }
+    });
+  }
+
+  @Override
+  public List<LtiJob> listJobs(String seriesId) {
+    final User user = securityService.getUser();
+    final EventSearchQuery query = new EventSearchQuery(securityService.getOrganization().getId(), user)
+            .withCreator(user.getName()).withSeriesId(StringUtils.trimToNull(seriesId));
+    try {
+      SearchResult<Event> results = this.searchIndex.getByQuery(query);
+      ZonedDateTime startOfDay = ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS);
+      return Arrays.stream(results.getItems())
+              .map(SearchResultItem::getSource)
+              .filter(e -> ZonedDateTime.parse(e.getCreated()).compareTo(startOfDay) > 0)
+              .map(e -> new LtiJob(e.getTitle(), e.getDisplayableStatus(workflowService.getWorkflowStateMappings())))
+              .collect(Collectors.toList());
+    } catch (SearchIndexException e) {
+      throw new RuntimeException("search index exception", e);
+    }
+  }
+
+  @Override
+  public void upsertEvent(
+          final LtiFileUpload file,
+          final String captions,
+          final String eventId,
+          final String seriesId,
+          final String metadataJson) throws UnauthorizedException, NotFoundException {
+    if (eventId != null) {
+      updateEvent(eventId, metadataJson);
+      return;
+    }
+    if (workflow == null || workflowConfiguration == null) {
+      throw new RuntimeException("no workflow configured, cannot upload");
+    }
+    try {
+      MediaPackage mp = ingestService.createMediaPackage();
+      if (mp == null) {
+        throw new RuntimeException("Unable to create media package for event");
+      }
+      if (captions != null) {
+        final MediaPackageElementFlavor captionsFlavor = new MediaPackageElementFlavor("vtt+en", "captions");
+        final MediaPackageElementBuilder elementBuilder = MediaPackageElementBuilderFactory.newInstance().newElementBuilder();
+        final MediaPackageElement captionsMpe = elementBuilder
+                .newElement(MediaPackageElement.Type.Attachment, captionsFlavor);
+        captionsMpe.setMimeType(mimeType("text", "vtt"));
+        captionsMpe.addTag("lang:en");
```

Comment:
That was basically the question I was asking myself as well. At the time of writing, the whole subtitle stuff was not very well documented, so I tried stuff out until paella/theodul actually displayed the subtitles.

Reply:
Considering Plapadoo's current state, I would be tempted to file this as an issue once this PR is in.  This will come back and bite someone if we don't fix it at some point.